client/modules/IDE/components/FileNode.jsx

```diff
@@ -230,6 +230,45 @@ class FileNode extends React.Component {
     this.props.hideFolderChildren(this.props.id);
   };
 
+  handleDrag = (e) => {
+    e.dataTransfer.setData('text', `${this.props.id} ${this.props.parentId}`);
+  };
+
+  handleDragOver = (e) => {
+    e.preventDefault();
+    if (this.props.fileType === 'folder') {
+      e.target.parentNode.parentNode.style.border = '1px solid #ed225d';
+    } else if (this.props.fileType === 'file') {
+      e.target.parentNode.parentNode.parentNode.parentNode.parentNode.style.border =
+        '1px solid #ed225d';
+    }
+  };
+
+  handleDragLeave = (e) => {
+    e.preventDefault();
+    if (this.props.fileType === 'folder') {
+      e.target.parentNode.parentNode.style.border = null;
+    } else if (this.props.fileType === 'file') {
+      e.target.parentNode.parentNode.parentNode.parentNode.parentNode.style.border = null;
```

Comment:
We want to avoid direct manipulation of styles and instead set some state somewhere which controls the style.  I’m also concerned that the number of parent levels that we need to go up by might depend on where this specific file I in the tree, and could be incorrect if it’s a few levels deep.